packages/api/cms-api/src/user-permissions/access-control.service.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable } from "@nestjs/common";
+
+import { CurrentUserInterface } from "../auth/current-user/current-user";
+import { CurrentUserContentScope } from "./dto/current-user";
+import { ContentScope } from "./interfaces/content-scope.interface";
+import { Permission } from "./interfaces/user-permission.interface";
+import { PermissionConfiguration } from "./user-permissions.types";
+
+export type AllowedPermission =
+    | {
+          permission: keyof Permission;
+          configuration?: PermissionConfiguration;
+      }
+    | keyof Permission;
+
+@Injectable()
+export class AccessControlService {
+    canAccessScope(requestScope: ContentScope, user: CurrentUserInterface): boolean {
+        return this.isAllowed(user, "pageTree", requestScope);
+    }
+
+    isAllowed(user: CurrentUserInterface, permission: AllowedPermission, contentScope?: ContentScope): boolean {
+        return this._isAllowed(user, permission, contentScope);
+    }
+
+    _isAllowed(user: CurrentUserInterface, permission: AllowedPermission, contentScope?: ContentScope): boolean {
```

Comment:
Why do you need this additional private method?

Reply:
It's possible to extend this service (https://github.com/vivid-planet/comet/pull/1223/files#diff-87c5a6413e447c9c46e24a82309ca67e8200fd20332af148f99c70344145a287R19) and this gives the possibility to overwrite the isAllowed-method while still being able to use the logic for checking rights.

Reply:
you could use super.isAllowed to access the parent method

Reply:
https://github.com/vivid-planet/comet/pull/1223/commits/e076e78b5c6112b31cbc2ed66c3c7a50c9b3603b